src/utils.rs

```diff
@@ -0,0 +1,26 @@
+use alloy::primitives::{PrimitiveSignature, B256};
+
+pub fn extract_signature_components(sig: &PrimitiveSignature) -> (u8, B256, B256) {
```

Comment:
let's add doc comments as to why this is needed, and what this does. (`///`)

Reply:
done!